src/sagemaker_inference/transformer.py

```diff
@@ -224,13 +243,36 @@ def _default_transform_fn(self, model, input_data, content_type, accept):
             input_data (obj): the request data.
             content_type (str): the request content type.
             accept (str): accept header expected by the client.
+            context (obj): the reqeust context.
 
         Returns:
             obj: the serialized prediction result or a tuple of the form
                 (response_data, content_type)
 
         """
-        data = self._input_fn(input_data, content_type)
-        prediction = self._predict_fn(data, model)
-        result = self._output_fn(prediction, accept)
+        data = self._run_handler_function(self._input_fn, *(input_data, content_type))
+        prediction = self._run_handler_function(self._predict_fn, *(data, model))
+        result = self._run_handler_function(self._output_fn, *(prediction, accept))
+        return result
+
+    def _run_handler_function(self, func, *argv):
+        """Helper to call the handler function which covers 2 cases:
+        1. the handle function takes context
+        2. the handle function does not take context
+        """
+        num_func_input = len(signature(func).parameters)
+        if num_func_input == len(argv):
+            # function does not take context
+            result = func(*argv)
+        elif num_func_input == len(argv) + 1:
```

Comment:
why do we assume that the extra argument is always context? Here we are not performing any sort of checks to ensure that user is passing context to the `_fn` ?

Reply:
Also, is this scalable? What if the optional arguments change (or increase) ?

Reply:
We are not taking arguments from customers. Rather, we are passing `context` from the hander service to customer so that they can use it to select multi GPUs. This approach is also recommended by [torchserve](https://github.com/pytorch/serve/blob/master/docs/custom_service.md#handling-model-execution-on-multiple-gpus). Basically this is a contract we offer to the customer and it should always be context